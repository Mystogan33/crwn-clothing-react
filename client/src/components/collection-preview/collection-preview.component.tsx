import React, { FC } from 'react';
import { withRouter, RouteComponentProps } from 'react-router-dom';

import {
  CollectionPreviewContainer,
  TitleContainer,
  PreviewContainer
} from './collection-preview.styles';

import { CollectionItem } from '../collection-item/collection-item.component';
import { ICollection } from '../../interfaces/interfaces';

type CollectionPreviewProps = RouteComponentProps & ICollection;

export const CollectionPreviewComponent: FC<CollectionPreviewProps> = ({ title, items, routeName, match: { path } }) => {
  const renderedfilteredItems = items
    .filter((_, idx) => idx < 4)
    .map(item => <CollectionItem key={item.id} item={item} />);
  
  return (
    <CollectionPreviewContainer>
      <TitleContainer to={`${path}/${routeName}`}>{title.toUpperCase()}</TitleContainer>
      <PreviewContainer>
        { renderedfilteredItems }
      </PreviewContainer>
    </CollectionPreviewContainer>
  );
};

export const CollectionPreview = withRouter(CollectionPreviewComponent);
