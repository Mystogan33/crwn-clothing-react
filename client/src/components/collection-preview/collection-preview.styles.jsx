import styled from 'styled-components';
import { Link } from 'react-router-dom';

const CollectionPreviewContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 30px;
`;

const TitleContainer = styled(Link)`
  font-size: 28px;
  margin-bottom: 25px;
  cursor: pointer;
  &:hover {
    color: grey;
  }
`;

const PreviewContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

export {
  CollectionPreviewContainer,
  TitleContainer,
  PreviewContainer
};
