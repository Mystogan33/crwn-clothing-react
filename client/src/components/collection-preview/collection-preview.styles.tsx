import styled from 'styled-components';
import { Link } from 'react-router-dom';

const CollectionPreviewContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 30px;

  @media screen and (max-width: 800px) {
    align-items: center;
  }
`;

const TitleContainer = styled(Link)`
  font-size: 28px;
  margin-bottom: 30px;
  cursor: pointer;

  &:hover {
    color: grey;
  }
`;

const PreviewContainer = styled.div`
  display: flex;
  justify-content: space-between;

  @media screen and (max-width: 800px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 30px;
  }
`;

export {
  CollectionPreviewContainer,
  TitleContainer,
  PreviewContainer
};
