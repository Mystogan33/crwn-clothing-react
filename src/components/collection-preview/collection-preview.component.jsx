import React from 'react';
import './collection-preview.styles.scss';

import { CollectionItem } from '../index';

export const CollectionPreview = ({title, items}) => (
  <div className="collection-preview">
    <h1>{title.toUpperCase()}</h1>
    <div className="preview">
      {
        items
        .filter((item, idx) => idx < 4)
        .map((item) => (
          <CollectionItem key={item.id} item={item} />
        ))
      }
    </div>
  </div>
);
