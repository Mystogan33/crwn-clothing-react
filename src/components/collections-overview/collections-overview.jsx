import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectShopCollections } from '../../redux/shop/shop.selectors';

import { CollectionPreview } from '../index';
import './collections-overview.styles.scss';

const mapStateToProps = createStructuredSelector({
  collections: selectShopCollections
});

export const CollectionsOverview = connect(mapStateToProps)(
  ({collections}) => (
    <div className="collections-overview">
      {
        collections.map(({id, ...otherProps }) => (
          <CollectionPreview key={id} {...otherProps} />
        ))
      }
    </div>
  )
);
