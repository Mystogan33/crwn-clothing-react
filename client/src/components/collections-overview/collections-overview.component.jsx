import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectShopCollectionsForPreview } from '../../redux/shop/shop.selectors';

import { CollectionPreview } from '../collection-preview/collection-preview.component';
import { CollectionsOverviewContainer } from './collections-overview.styles';

export const CollectionsOverview = ({collections}) => (
  <CollectionsOverviewContainer>
    { collections.map(({id, ...otherProps }) => (
        <CollectionPreview key={id} {...otherProps} />
      ))
    }
  </CollectionsOverviewContainer>
);

// export const CollectionsOverviewComponent = ({collections}) => (
//   <CollectionsOverviewContainer>
//     { collections.map(({id, ...otherProps }) => (
//         <CollectionPreview key={id} {...otherProps} />
//       ))
//     }
//   </CollectionsOverviewContainer>
// );

// const mapStateToProps = createStructuredSelector({
//   collections: selectShopCollectionsForPreview
// });

// export const CollectionsOverview = connect(mapStateToProps)(CollectionsOverviewComponent);
