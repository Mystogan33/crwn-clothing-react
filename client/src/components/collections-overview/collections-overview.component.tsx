import React from 'react';

import { CollectionPreview } from '../collection-preview/collection-preview.component';
import { CollectionsOverviewContainer } from './collections-overview.styles';

export const CollectionsOverview = ({ collections }) => (
  <CollectionsOverviewContainer>
    { collections.map(({id, ...otherProps }) => (
        <CollectionPreview key={id} {...otherProps} />
      ))
    }
  </CollectionsOverviewContainer>
);