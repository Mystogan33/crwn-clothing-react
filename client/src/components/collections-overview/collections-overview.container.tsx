import React, { FC } from 'react';
import { connect, ConnectedProps } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectIsCollectionFetching, selectShopCollectionsForPreview } from '../../redux/shop/shop.selectors';

import { CollectionPreview } from '../collection-preview/collection-preview.component';
import { WithSpinner } from '../with-spinner/with-spinner.component';
import { CollectionsOverviewContainer } from './collections-overview.styles';

const mapStateToProps = createStructuredSelector({
  isLoading: selectIsCollectionFetching,
  collections: selectShopCollectionsForPreview
});

const connector = connect(mapStateToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;
type ICollectionsOverviewProps = PropsFromRedux;

export const CollectionsOverviewCmp: FC<ICollectionsOverviewProps> = ({ collections }) => (
  <CollectionsOverviewContainer>
    { collections.map(({id, ...otherProps }) => (
        <CollectionPreview key={id} {...otherProps} />
      ))
    }
  </CollectionsOverviewContainer>
);

const connectedComponent = connector(CollectionsOverviewCmp);
const CollectionsOverview = WithSpinner(connectedComponent);
export default CollectionsOverview;
