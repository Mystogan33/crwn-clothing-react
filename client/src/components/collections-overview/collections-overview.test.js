import React from 'react';
import { shallow } from 'enzyme';
import { CollectionsOverviewCmp } from './collections-overview.component';
import { CollectionPreview } from '../collection-preview/collection-preview.component';

describe('CollectionsOverview component', () => {
  let wrapper;
  let mockCollections;

  beforeEach(() => {

    mockCollections = [
      {
        id: 1,
        title: 'MockCollection1',
        items: [
          {
            id: 998,
            imageUrl: 'https://testimage.com',
            name: 'MockItem 1',
            price: 325
          }
        ]
      },
      {
        id: 2,
        title: 'MockCollection2',
        items: [
          {
            id: 999,
            imageUrl: 'https://testimage.com',
            name: 'MockItem 1',
            price: 325
          }
        ]
      }
    ];

    wrapper = shallow(<CollectionsOverviewCmp collections={mockCollections} />);
  });

  
  it('should render CollectionsOverview component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render number of CollectionPreview equal to collections prop length', () => {
    expect(wrapper.exists(CollectionPreview)).toBe(true);
    expect(wrapper.find(CollectionPreview).length).toEqual(mockCollections.length);
  });

  it('should not render any CollectionPreview if collections prop is empty', () => {
    const newWrapper = shallow(<CollectionsOverviewCmp collections={[]} />);
    expect(newWrapper.exists(CollectionPreview)).toBe(false);
  });
});
