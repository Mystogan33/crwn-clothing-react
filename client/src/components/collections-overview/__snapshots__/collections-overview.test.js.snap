// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsOverview component should render CollectionsOverview component 1`] = `ShallowWrapper {}`;
